Classes/Service/OgRendererService.php

```diff
@@ -140,7 +140,18 @@ public function main($content, $conf)
             $og['image'] = $fileObjects;
 
         // Get url
-        $og['url'] = htmlentities(GeneralUtility::getIndpEnv('TYPO3_REQUEST_URL'));
+        $og['url'] = GeneralUtility::getIndpEnv('TYPO3_SITE_URL');
+        $cObject = GeneralUtility::makeInstance('tslib_cObj');
+        $configurations['additionalParams'] = "&L=".(int)GeneralUtility::_GP('L');
+        $configurations['returnLast'] = 'url'; // get it as URL
+        $configurations['parameter'] =  $GLOBALS['TSFE']->id;
```

Comment:
What would be a correct way to implement this? Having different URLs is not good for SEO.